fwoa-tools/src/migrationImport.ts

```diff
@@ -150,7 +142,83 @@ async function startImport(folderNames: string[]): Promise<void> {
   }
 }
 
-async function verifyFolderImport(folderName: string, s3Uri: string): Promise<void> {
+async function checkFolderSizeOfResource(resources: string[]): Promise<void> {
+  const s3Client = new S3({
+    region: API_AWS_REGION!
+  });
+  for (let i = 0; i < resources.length; i++) {
+    // eslint-disable-next-line security/detect-object-injection
+    const resource = resources[i];
+    // We don't check Binary resource ndjson file, instead we check the `Binary_converted` ndjson files
+    // Each binary file is limited to 5GB and each `Binary_converted` ndjson file has only one binary file
+    if (resource === 'Binary') {
+      continue;
+    }
+    console.log(`Checking resource ${resource}`);
+    const maximumFolderSize = 536870912000; // 500 GB in bytes
+
+    let folderSize = 0;
+    let continuationToken: string | undefined = undefined;
+    do {
+      const response: ListObjectsV2Output = await s3Client
+        .listObjectsV2({
+          Bucket: IMPORT_OUTPUT_S3_BUCKET_NAME!,
+          Prefix: `${outputFile.jobId}/${resource}`,
+          ContinuationToken: continuationToken
+        })
+        .promise();
+      if (response.Contents) {
+        response.Contents.forEach((item) => {
+          if (item.Size) {
+            folderSize += item.Size;
+          }
+        });
+      }
+      continuationToken = response.NextContinuationToken;
+    } while (continuationToken);
+    if (folderSize > maximumFolderSize) {
+      throw new Error(
+        `Resource ${resource} has a folder size of ${folderSize} bytes, which is larger than the maximum folder size of ${maximumFolderSize} bytes`
+      );
+    } else {
+      console.log(
+        `${resource} folder size is ${folderSize} bytes. This is within maximum folder size limit.`
+      );
+    }
+  }
+}
+
+async function checkConvertedBinaryFileSize(): Promise<void> {
+  console.log('Checking Binary file size');
+  const s3Client = new S3({
+    region: API_AWS_REGION!
+  });
+  const maximumBinaryFileSize = 5368709120; // 5 GB in bytes
+  const convertedBinaryFolderName = 'Binary_converted';
+  let continuationToken: string | undefined = undefined;
+  do {
+    const response: ListObjectsV2Output = await s3Client
+      .listObjectsV2({
+        Bucket: IMPORT_OUTPUT_S3_BUCKET_NAME!,
+        Prefix: `${outputFile.jobId}/${convertedBinaryFolderName}`,
```

Comment:
Ah yes, good question. I just checked and the code should still work since `Binary_coverted_1/Binary-1.ndjson` still has the prefix of `Binary_converted`.